jscomp/core/lam_compile_external_call.ml

```diff
@@ -312,18 +312,24 @@ let translate_ffi (cxt : Lam_compile_context.t) arg_types
             (let self = translate_scoped_access js_send_scopes self in
              if dynamic then splice_obj_fn_apply self name args
              else
-               E.call
-                 ~info:{ arity = Full; call_info = Call_na }
-                 (E.dot self name) args)
+               match new_ with
+               | true -> E.new_ (E.dot self name) args
+               | false ->
+                   E.call
+                     ~info:{ arity = Full; call_info = Call_na }
+                     (E.dot self name) args)
         else
           let args, self = List.split_at_last args in
           let arg_types, _ = List.split_at_last arg_types in
           let args, eff = assemble_args_no_splice arg_types args in
           add_eff eff
             (let self = translate_scoped_access js_send_scopes self in
-             E.call
-               ~info:{ arity = Full; call_info = Call_na }
-               (E.dot self name) args)
+             match new_ with
+             | true -> E.new_ (E.dot self name) args
+             | false ->
+                 E.call
+                   ~info:{ arity = Full; call_info = Call_na }
+                   (E.dot self name) args)
```

Comment:
Do you think it's worth to abstract this `match` into a function? The same code appears 4 times.

Reply:
absolutely